### PR TITLE
Fail node only if all monitored services are down

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

**Comment by MJPIPER99** [comment]
All down . mission FAILURE. Back to restart.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java [hunk 1]
```diff
@@ -169,27 +170,38 @@ private boolean failAllowedFor(NodeType nodeType) {
     }
 
     /**
-     * If the node is positively DOWN, and there is no "down" history record, we add it.
-     * If the node is positively UP we remove any "down" history record.
+     * If a node remains bad for a long time, the NodeFailer will eventually try to fail the node.
```
Comment (bratseth) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java]: I think this JavaDoc should explain what this method returns instead,

**Review by bratseth — APPROVED** [review]
LGTM

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java]: Added a line for that.

**Pull request merged**